File: 2718.py
casos_teste = int(input(""))

for i in range(casos_teste):
    lampadas = maior_sequencia = 0
    numero = int(input(""))
    
    while (numero / 2 > 0):
        if (numero % 2 == 1):
            lampadas += 1
            if (lampadas > maior_sequencia):
                maior_sequencia = lampadas
        else:
            lampadas = 0
        numero //= 2
    
    if (lampadas + 1 > maior_sequencia and numero > 0):
        maior_sequencia = lampadas + 1
    
    print(f'{maior_sequencia}')

File: 2718.c
#include <stdio.h>
 
int main() {
    int casos_teste;
    
    scanf("%d", &casos_teste);
    
    for (int i = 0; i < casos_teste; i++) {
        long long int numero;
        int lampadas = 0, maior_sequencia = 0;
        
        scanf("%lld", &numero);
        
        while (numero / 2) {
            if (numero % 2) {
                lampadas++;
                if (lampadas > maior_sequencia)
                    maior_sequencia = lampadas;
            }
            else lampadas = 0;
            
            numero /= 2;
        }

        if ((lampadas + 1) > maior_sequencia && numero > 0) 
            maior_sequencia = ++lampadas;
        
        printf("%d\n", maior_sequencia);
    }
 
    return 0;
}
